deploy/crds/core_v1_storagecluster_crd.yaml

```diff
@@ -39,6 +39,18 @@ spec:
       openAPIV3Schema:
         type: object
         properties:
+          apiVersion:
+            type: string
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+          kind:
+            type: string
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+          metadata:
```

Comment:
In case of a stricter validation, won't it complain for the fields under metadata (name, namespace, resourceVersion, etc)?

Reply:
So I checked the CRDs for some other popular projects and they seem to not enforce this as well. For example:
- https://github.com/cert-manager/cert-manager/blob/master/deploy/crds/crd-challenges.yaml#L66
- https://github.com/metallb/metallb/blob/main/charts/metallb/charts/crds/templates/crds.yaml#L41

The reason it does not complain has to do with a design choice in the script used to convert a CRD into a json schema definition: https://github.com/yannh/kubeconform/blob/master/scripts/openapi2jsonschema.py#L23 . There is a boolean field called "additionalProperties" which determines whether you can have additional fields (in this case, in metadata) beyond those that are defined. However, if there are no "properties" to begin with, then the field is not added (as False). I guess the strict validation is only done when there is a "properties" and/or "required" field.

If I were to add those fields, it would look something like this:
https://github.com/yannh/kubernetes-json-schema/blob/master/v1.27.6-standalone-strict/pod-v1.json#L24-L270 (but in yaml instead of json)
Based on the docs: https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/